### ISSUE TITLE
Making lowliner better understand data frames: map_rows()

### ISSUE THREAD

**Issue by lionel-** [hadley/purrr#44]
This is a version of `map_n()` that returns a data frame. If the option `.trace` is set (it is by default), the output is returned col-binded to the original data frame. This way we address @baptiste's concerns in hadley/dplyr#441.

If `.f` does not return a data frame but a bare atomic vector of length n, it is coerced to a data frame of n columns. In other cases, the ouput is coerced to a one-row list-column.

In action:
```{r}
p <- expand.grid(mean = 1:5, sd = seq(0, 1, length = 10))
# Next line is equivalent to: p %>% plyr::mdply(rnorm, n = 5) %>% tbl_df()
p %>% map_rows(rnorm, n = 5)
```
```{r}
## Source: local data frame [50 x 7]
## 
##    mean      sd       1      2       3       4      5
## 1     1 0.00000 1.00000 1.0000 1.00000 1.00000 1.0000
## 2     2 0.00000 2.00000 2.0000 2.00000 2.00000 2.0000
## 3     3 0.00000 3.00000 3.0000 3.00000 3.00000 3.0000
## 4     4 0.00000 4.00000 4.0000 4.00000 4.00000 4.0000
## 5     5 0.00000 5.00000 5.0000 5.00000 5.00000 5.0000
## 6     1 0.11111 0.98537 1.0664 0.95703 0.80309 0.9408
## 7     2 0.11111 1.91233 2.0431 1.93546 1.76423 1.7954
## 8     3 0.11111 2.89180 3.1357 2.97548 3.04977 3.0097
## 9     4 0.11111 4.09118 4.0426 3.95294 4.02712 4.0887
## 10    5 0.11111 5.02453 4.8226 5.02239 5.07338 5.0010
## ..  ...     ...     ...    ...     ...     ...    ...
```

```{r}
p <- data_frame(
  a = rerun(5, rnorm(10, 5)),
  b = rerun(5, rnorm(10))
)
p %>% map_rows(function(a, b) lm(a ~ b))
```
```{r}
## Source: local data frame [5 x 3]
## 
##           a         b     out
## 1 <dbl[10]> <dbl[10]> <S3:lm>
## 2 <dbl[10]> <dbl[10]> <S3:lm>
## 3 <dbl[10]> <dbl[10]> <S3:lm>
## 4 <dbl[10]> <dbl[10]> <S3:lm>
## 5 <dbl[10]> <dbl[10]> <S3:lm>
```

The code is in https://gist.github.com/lionel-/0b41c6b9d3554725807a

**Comment by lionel-** [comment]
> a bare atomic vector of length n is coerced to a data frame of n columns.

This was to reproduce `plyr::mdply` behavior but it may be better to return a tidy dataframe instead.

**Comment by lionel-** [comment]
Closed by #51
